processing/src/main/java/io/druid/segment/StringDimensionIndexer.java

```diff
@@ -270,6 +270,18 @@ public StringDimensionIndexer(MultiValueHandling multiValueHandling, boolean has
     return encodedDimensionValues;
   }
 
+  @Override
+  public long estimateEncodedKeyComponentSize(int[] key)
+  {
+    // string length is being accounted for each time they are referenced, even though they are stored just once
```

Comment:
i guess it was just to leave more buffer.

Reply:
When we talked about the design of this, I think we found that it was simpler to count the strings each time they're referenced, given how the dimension handler interfaces worked. And nothing really awful should happen: it might just overestimate somewhat and spill a bit too often. But if you are going to err, that's the right side to err on.